src/System.Private.CoreLib/src/System/Environment.cs

```diff
@@ -50,6 +50,20 @@ internal static long TickCount64
             }
         }
 
+        public static string MachineName
+        {
+            get
+            {
+                const int MaxMachineNameLength = 256;
+                StringBuilder buf = new StringBuilder(MaxMachineNameLength);
+                int capacity = MaxMachineNameLength;
+                if (Interop.mincore.gethostname(buf, capacity) != 0)
```

Comment:
Looks like there's a mismatch between the `MaxMachineNameLength` and `StringBuilder` capacity here: `gethostname` is expecting the length in bytes, but the `StringBuilder` capacity is the length of characters. Also, `StringBuilder` marshaling is expensive. If `StringBuilder` needs to be used here, at the very least it should have an `[Out]` attribute specified, but why can't this just be `byte*` and called with a stackallocated array? See https://github.com/dotnet/corefx/pull/4127